### PR TITLE
✨ activation infos and Workflow Activation Trigger

### PR DESCRIPTION
**Pull request by lublak**
Allow trigger on n8n start

### CODE REVIEW

**Comment by lublak** [comment]
Currently it triggers also if a workflow get saved an restarted, can be also usefull but this should be a other node.
I fix it.

File: packages/nodes-base/nodes/N8nStartTrigger.node.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	};
+
+
+	async trigger(this: ITriggerFunctions): Promise<ITriggerResponse> {
+		const self = this;
+		async function manualTriggerFunction() {
+			self.emit([self.helpers.returnJsonArray([{}])]);
+		}
+		
+		manualTriggerFunction();
```
Comment (lublak) [inline packages/nodes-base/nodes/N8nStartTrigger.node.ts:32]: Only call this function at the start.

**Comment by lublak** [comment]
I thin to get this type of trigger worked we need some information if the node gets started from init:

https://github.com/n8n-io/n8n/blob/master/packages/cli/src/ActiveWorkflowRunner.ts#L68

So some .activationMode can be usefull with: 

```ts
export type WorkflowActivationMode = 'init' | 'update' | 'activate';
```

init -> for activation at start
update -> for activation if workflow changed
activate -> if workflow activate

@janober do you have any comments about it?

**Comment by lublak** [comment]
last fix needed i hope (on manual execute):

```
UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:24532) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:30484) UnhandledPromiseRejectionWarning: Error: Overwrite NodeExecuteFunctions.getExecuteTriggerFunctions.emit function!
```

**Comment by janober** [comment]
Sorry that it took that long. We now thought about it, reviewed, made some minor changes, and then merged with #1570

Thanks a lot for that great contribution!

Will be released with the next version.

**Comment by janober** [comment]
Got released with n8n@0.113.0

**Pull request closed without merging**